### PR TITLE
Cleanup and optimise interpolation code

### PR DESCRIPTION
**Pull request by nyalldawson**
A bunch of modernisations and optimisations for the interpolation code

### CODE REVIEW

File: src/analysis/interpolation/qgstininterpolator.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ QgsTINInterpolator::~QgsTINInterpolator()
   delete mTriangleInterpolator;
 }
 
-int QgsTINInterpolator::interpolatePoint( double x, double y, double &result )
+int QgsTINInterpolator::interpolatePoint( double x, double y, double &result, QgsFeedback * )
```
Comment (m-kuhn) [inline src/analysis/interpolation/qgstininterpolator.cpp]: QgsTinInterpolator?

File: src/analysis/interpolation/qgsinterpolator.h [hunk 2]
```diff
@@ -22,39 +22,76 @@
...
+class QgsFeatureSource;
 class QgsGeometry;
+class QgsFeedback;
 
-struct ANALYSIS_EXPORT vertexData
+/**
+ * Interpolation data for an individual source vertex.
+ * \since QGIS 3.0
+ */
+struct ANALYSIS_EXPORT QgsInterpolatorVertexData
```
Comment (m-kuhn) [inline src/analysis/interpolation/qgsinterpolator.h]: Do we have no class in core for this?

**Review by m-kuhn — APPROVED** [review]
Just some minor thoughts

File: src/analysis/interpolation/qgsinterpolator.h (hunk 2 above)
Comment (nyalldawson) [reply src/analysis/interpolation/qgsinterpolator.h]: Well... we could use QgsPoint. But I was concerned about the weight of it (extra double, vtable). I'll make this non-public API though.

File: src/analysis/interpolation/qgstininterpolator.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/analysis/interpolation/qgstininterpolator.cpp]: yep :)

more fixes inbound!

**Pull request merged**